pandas/core/generic.py

```diff
@@ -1501,6 +1501,13 @@ def bool(self) -> bool_t:
         >>> pd.DataFrame({'col': [False]}).bool()
         False
         """
+
+        warnings.warn(
+            "NDFrame.bool is now deprecated and will be removed in future releases "
```

Comment:
NDFrame is internal, can you replace with `type(self).__name__`. This will resolve to DataFrame or Series as appropriate.

Reply:
I think it's sufficient to say "and will be removed in a future version of pandas".

Reply:
actually if i replace `NDFrame` with `type(self).__name__` then how should I change the entry in conftest because there i don't know how to go about it

Reply:
I think the entry in conftest is regex. Also I believe it will work with partial matches, so you could leave the Series / DataFrame part out - but double check I am right here.